src/main/scala/PropertyChecker.scala

```diff
@@ -630,4 +664,285 @@ object PropertyChecker {
       }
     }
   }
+
+  def uriTemplateProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      value match {
+          // Don't know how to place a URI object in JsonNode, keeping the text value as of now
+        case s: TextNode => (s, Array[String](), csvwPropertyType)
+        case _ => (new TextNode(""), Array[String](PropertyChecker.invalidValueWarning), csvwPropertyType)
+      }
+    }
+  }
+
+  def textDirectionProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      value match {
+        case s: TextNode if Array[String]("ltr", "rtl", "inherit").contains(s.asText()) => {
```

Comment:
Does the CSVW spec say anything about supporting case-insensitive matches of these strings?

Reply:
I didn't find anything on case-insensitive matches of these strings. Trusting the ruby implementation here